src/main/java/org/cyclops/integrateddynamics/core/network/EnergyChannel.java

```diff
@@ -0,0 +1,90 @@
+package org.cyclops.integrateddynamics.core.network;
+
+import org.cyclops.integrateddynamics.GeneralConfig;
+import org.cyclops.integrateddynamics.api.network.IChanneledNetwork;
+import org.cyclops.integrateddynamics.api.network.IPositionedAddonsNetwork.PrioritizedPartPos;
+
+import net.minecraftforge.energy.IEnergyStorage;
+
+public class EnergyChannel implements IEnergyStorage {
+    final EnergyNetwork network;
+    final int channel;
+
+    EnergyChannel(EnergyNetwork network, int channel) {
```

Comment:
Similar question here, any reason why this constructor is package-private and not just public?

Reply:
That's true, but I'm just thinking in terms of extensibility. If some external mod would like to create its own energy channel for whatever reason, should it be able to use this implementation? I think the answer is yes, unless something could go wrong that way?